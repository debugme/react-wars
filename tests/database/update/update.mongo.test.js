import _ from 'lodash'
import path from 'path'
import dotenv from 'dotenv'
import mongoose from 'mongoose'

import Character from '../../../source/database/models/Character'
import fields from './update.data.json'

describe('update existing document in characters collection', () => {

  beforeAll((done) => {
    mongoose.Promise = global.Promise
    dotenv.config({ path: path.resolve(__dirname, '..', '..', '.env') })
    const saveCharacter = () => new Character(fields).save()
    mongoose.connect(process.env.mongodbUri)
      .then(saveCharacter)
      .then(done)
  })

  afterAll((done) => {
    Character.collection.remove()
      .then(() => mongoose.connection.close())
      .then(done)
  })

  it('should update a character if correct values are provided', (done) => {
    const match = { name: fields.name }
    const update = { hair_color: 'black'}
    const options = { new : true }
    const keys = Object.keys(fields)
    const check = (data) => expect(_.pick(data, keys)).toMatchSnapshot()
    Character.findOneAndUpdate(match, update, options)
      .then(check)
      .then(done)
  })

})