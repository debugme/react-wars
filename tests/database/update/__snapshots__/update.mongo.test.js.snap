// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`update existing document in characters collection should update a character if correct values are provided 1`] = `
Object {
  "birth_year": "29BBY",
  "eye_color": "brown",
  "hair_color": "black",
  "height": 180,
  "is_male": true,
  "mass": 80,
  "name": "Han Solo",
  "skin_color": "fair",
}
`;
